### PR TITLE
Remove multiple lodash methods from the codebase

### PR DESCRIPTION
**Pull request by perunt**
Fixes RNBW-3856

## What changed (plus any additional context for devs)
Replaced lodash `isNil, isEmpty, isString, isNull, isUndefined, isArray, isNumber` on the native one

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels?
- [ ] Added e2e tests? if not please specify why
- [ ] If you added new files, did you update the CODEOWNERS file?

### CODE REVIEW

**Review by tchayen — APPROVED** [review]
I was so scared when I saw '74 files changed' but fortunately the changes are super straightforward. Good job! 😄

File: src/helpers/utilities.ts [hunk 1]
```diff
@@ -1,10 +1,24 @@
...
+export const isEmpty = (obj: any): boolean =>
+  [Object, Array].includes((obj || {}).constructor) &&
+  !Object.entries(obj || {}).length;
+export const isString = (str: any): str is string => {
+  if (str != null && typeof str.valueOf() === 'string') {
+    return true;
+  }
+  return false;
+};
+export const isNumber = (value: any) => typeof value === 'number';
```
Comment (terrysahaidak) [inline src/helpers/utilities.ts:20]:
```suggestion
export const isNumber = (value: any) => typeof value === 'number' && !Number.isNaN(value);
```

Comment (perunt) [reply src/helpers/utilities.ts:20]: I compared it to lodash and missed this case, thanks

**Review by estebanmino — APPROVED** [review]
LGTM

**Review by brunobar79 — APPROVED** [review]

**Review by osdnk — APPROVED** [review]

File: src/react-native-animated-charts/Example/.prettierrc.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 module.exports = {
-  bracketSpacing: false,
-  jsxBracketSameLine: true,
+  arrowParens: 'avoid',
+  bracketSpacing: true,
+  quoteProps: 'consistent',
   singleQuote: true,
-  trailingComma: 'all',
+  trailingComma: 'es5',
```
Comment (estrattonbailey) [inline src/react-native-animated-charts/Example/.prettierrc.js:6]: We should make this change as a separate PR

**Comment by terrysahaidak** [comment]
This will be addressed in the future after some discussion
